mlflow/pipelines/steps/evaluate.py

```diff
@@ -394,7 +394,10 @@ def from_pipeline_config(cls, pipeline_config, pipeline_root):
         step_config["target_col"] = pipeline_config.get("target_col")
         if "positive_class" in pipeline_config:
             step_config["positive_class"] = pipeline_config.get("positive_class")
-        step_config["metrics"] = pipeline_config.get("metrics")
+        if pipeline_config.get("custom_metrics") is not None:
+            step_config["custom_metrics"] = pipeline_config["custom_metrics"]
```

Comment:
What's the reason to not just use `pipeline_config.get("custom_metrics")`

Reply:
I think we would ideally like to use step_config in the code rather than pipeline_config so we know the dependency for the step correctly since they are written to the individual step files as expected.

Reply:
No problem with putting everything in `step_config`. I meant that the effect of 
```
if pipeline_config.get("custom_metrics") is not None:
     step_config["custom_metrics"] = pipeline_config["custom_metrics"]
```
is almost identical to
`step_config["custom_metrics"] = pipeline_config.get("custom_metrics")`, if `custom_metrics` in `step_config` is accessed via `step_config.get("custom_metrics")`.

Reply:
Yes that makes sense. Addressed this comment.